vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -14,4 +16,10 @@ def scan
     render json: response_json, status: :forbidden
   end
 
+  def load_models
+    @user = User.find_by!(card_number: params[:card_id])
+    @control_device = ControlDevice.find(params[:controller_id])
+    @card_reader = CardReader.find(params[:reader_id])
```

Comment:
Since we have the foreign key, we should probably `@card_reader = @control_devise.card_readers.find` to scope it down a little better.